### ISSUE TITLE
Dynamic Circuit Error

### ISSUE THREAD

**Issue by bsiegelwax** [Qiskit-Partners/mapomatic#55]
I have a simple 2-qubit 2-bit dynamic circuit with 1 conditional rotation that executed on ibmq_manila just fine. But, when I add mapomatic I get:

CircuitError: "Unknown classical resource specifier: 'CircuitInstruction(operation=Instruction(name='rz', num_qubits=1, num_clbits=0, params=[-3.141592653589793]), qubits=(Qubit(QuantumRegister(2, 'q'), 1),), clbits=())'."

**Comment by taalexander** [comment]
Could you please provide an example Python snippet that completely reproduces the issue?

**Comment by bsiegelwax** [comment]
The entire circuit is:

q = QuantumRegister(1, name="q")
c = ClassicalRegister(1, name="c")
qc = QuantumCircuit(q, c, name="DVQE")
qc.ry(theta, 0)
qc.ry(-(pi/2), 0)
qc.measure(0, c)
with qc.if_test((c, 1)):
    qc.reset(0) # This line is actually indented.
    qc.ry(theta, 0) # This line is actually indented.
    qc.ry(-(pi/2), 0) # This line is actually indented.
qc.measure(0, c)

Again, that executes fine without mapomatic. It is transpiled at optimization_level=1. And, this is from this repository:

small_qc = mm.deflate_circuit(qc) # <---- The error is pointing here.
layouts = mm.matching_layouts(small_qc, backend)
scores = mm.evaluate_layouts(small_qc, layouts, backend)
qc = transpile(small_qc, backend, initial_layout=scores[0][0])

**Comment by nonhermitian** [comment]
The summary is that `mapomatic` does not really handle dynamic circuits at present.  I am pretty such for the case above it can be, I just need to figure out how to deflate and keep conditional information.  

A much more challenging issue is how to score dynamic circuits in general.  One could just assume every gate gets executed (worst case) and score that.  I am pretty sure that is how the current algorithm would parse it.

**Comment by mtreinish** [comment]
That is basically what we did inside terra for vf2postlayout when we made it control flow aware: https://github.com/Qiskit/qiskit-terra/commit/a01f738a56827e70e2df917cd4ec0b19f95707db

**Comment by nonhermitian** [comment]
Yeah, the only bit of work is that it seems that flow control instructions, eg `else_if` do not follow the same calling format as all the other instructions.
